plugins/nextcloud/nextcloud_

```diff
@@ -47,15 +50,23 @@ GPLv2
 
 =cut
 
+# shellcheck disable=SC1090
 . "$MUNIN_LIBDIR/plugins/plugin.sh"
 
-API_PATH="/ocs/v2.php/apps/serverinfo/api/v1/info?format=json"
+if [ "${MUNIN_DEBUG:-0}" = 1 ]; then
+    set -x
+fi
+
+API_PATH="${api_path:-/ocs/v2.php/apps/serverinfo/api/v1/info}?format=json"
 DOMAIN="${0##*nextcloud_}"
+TIMEOUT="${timeout:-2}"
+UPDATES_WARNING="${updates_warning:-1}"
 CLEANDOMAIN="$(clean_fieldname "$DOMAIN")"
 USERNAME="${username:-}"
 PASSWORD="${password:-}"
 
 print_json_data() {
+    # shellcheck disable=SC2039
     local FIRST="$1"
     [ -z "$FIRST" ] && exit 0
```

Comment:
This is unrelated to your current suggestions, but maybe you could check, whether `exit 0` is really the proper approach here (instead of `return 0`).

Reply:
It's a bit barbaric, but it looks like it's protecting against miscall from other places in the code. It kinda makes sense to use `exit` to terminate the script, rather than `return`, which would let it continue with weird values, but maybe it should actually be a non-zero exit status.

What do you think?

Reply:
> *barbaric*

:)

How about a message to stderr in order to not die silently?